packages/react-dom/src/shared/ReactDOMFloat.js

```diff
@@ -0,0 +1,27 @@
+import ReactDOMSharedInternals from '../ReactDOMSharedInternals';
+
+export function preinit() {
+  const dispatcher = ReactDOMSharedInternals.Dispatcher.current;
+  if (dispatcher) {
+    dispatcher.preinit.apply(this, arguments);
+  } else {
+    if (__DEV__) {
+      console.error(
+        'ReactDOM.preinit() was called when React did not expect it. This is a bug in React.',
```

Comment:
Isn't this just whenever you call outside of render on the server such as module init? That doesn't seem like a bug in React.

Reply:
One thing we could support but hooks doesn’t is pre* functions I’m layout effects. It doesn’t quite make sense you would want to call it there necessarily but it is currently the phase that styles are inserted in and a preinit could in theory insert synchronously if called that way. One could then argue we should support in passive effects but currently we don’t. What was the argument to not support preinit in effects? I feel like we talked about this months ago but things have changed a lot since then

Reply:
Yeah these should just be noop by default now